agent/consul/state/config_entry.go

```diff
@@ -704,6 +710,63 @@ func getReferencedProviderNames(j *structs.IntentionJWTRequirement, s []*structs
 	return providerNames
 }
 
+func validateJWTProviderIsReferenced(tx ReadTxn, kn configentry.KindName, ce structs.ConfigEntry) error {
+	meta := acl.NewEnterpriseMetaWithPartition(
+		kn.EnterpriseMeta.PartitionOrDefault(),
+		acl.DefaultNamespaceName,
+	)
+	entry, ok := ce.(*structs.JWTProviderConfigEntry)
+	if !ok {
+		return fmt.Errorf("invalid jwt provider config entry: %T", entry)
+	}
+
+	_, ixnEntries, err := configEntriesByKindTxn(tx, nil, structs.ServiceIntentions, &meta)
+	if err != nil {
+		return err
+	}
+
+	providerNames, err := collectJWTProviderNames(ixnEntries)
+	if err != nil {
+		return err
+	}
+
+	_, exist := providerNames[entry.Name]
```

Comment:
I can think of two possible improvements here.
1. Provide a list of intentions that reference the provider so the user can find them easily; OR
2. Use `entry.Name` in the loop over ixn providers so we can return early on name match (instead of building a map for N intentions)